cpp/src/replace/replace.cu

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2018 BlazingDB, Inc.
+ *     Copyright 2018 Cristhian Alberto Gonzales Castillo <cristhian@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thrust/device_ptr.h>
+#include <thrust/find.h>
+#include <thrust/execution_policy.h>
+
+#include "cudf.h"
+#include "utilities/error_utils.h"
+#include "dataframe/type_dispatcher.hpp"
+
+namespace{ //anonymous
+
+  constexpr int BLOCK_SIZE = 256;
+  
+  template <class T>
+  struct elements_are_equal{
+    __device__
+    elements_are_equal(T el) : element{el}
+    {
+    }
+
+    __device__
+    bool operator()(T other)
+    { 
+      return (cudf::detail::unwrap(other) == cudf::detail::unwrap(element));
```

Comment:
`unwrap` is no longer necessary because the `wrapper` structs now have appropriate arithmetic and boolean operators, this includes `operator==`.